src/frontend/data/stores/useLivePanelState/index.spec.ts

```diff
@@ -0,0 +1,270 @@
+      LivePanelItem.APPLICATION,
+      LivePanelItem.CHAT,
+    ]);
+  });
+
+  it('does not update the state changing selection with an invalid item', () => {
+    useLivePanelState.setState({
+      availableItems: [LivePanelItem.APPLICATION, LivePanelItem.CHAT],
+      currentItem: LivePanelItem.CHAT,
+      isPanelVisible: true,
+    });
+
+    //  update current item with invalid item
+    useLivePanelState
+      .getState()
+      .setPanelVisibility(true, LivePanelItem.JOIN_DISCUSSION);
+
+    expect(useLivePanelState.getState().isPanelVisible).toBe(true);
+    expect(useLivePanelState.getState().currentItem).toEqual(
+      LivePanelItem.CHAT,
+    );
+    expect(useLivePanelState.getState().availableItems).toEqual([
+      LivePanelItem.APPLICATION,
+      LivePanelItem.CHAT,
+    ]);
+  });
+
+  it('updates the state closing the panel', () => {
+    useLivePanelState.setState({
+      availableItems: [LivePanelItem.APPLICATION, LivePanelItem.CHAT],
+      currentItem: LivePanelItem.CHAT,
+      isPanelVisible: true,
+    });
+
+    //  update panel visibility
+    useLivePanelState.getState().setPanelVisibility(false);
+
+    expect(useLivePanelState.getState().isPanelVisible).toBe(false);
+    expect(useLivePanelState.getState().currentItem).toEqual(
+      LivePanelItem.CHAT,
+    );
+    expect(useLivePanelState.getState().availableItems).toEqual([
+      LivePanelItem.APPLICATION,
+      LivePanelItem.CHAT,
+    ]);
+  });
+
+  it('updates the state closing the panel and changing selection with a valid item', () => {
+    useLivePanelState.setState({
+      availableItems: [LivePanelItem.APPLICATION, LivePanelItem.CHAT],
+      currentItem: LivePanelItem.CHAT,
+      isPanelVisible: true,
+    });
+
+    //  update panel visibility with valid item to select
+    useLivePanelState
+      .getState()
+      .setPanelVisibility(false, LivePanelItem.APPLICATION);
+
+    expect(useLivePanelState.getState().isPanelVisible).toBe(false);
+    expect(useLivePanelState.getState().currentItem).toEqual(
+      LivePanelItem.APPLICATION,
+    );
+    expect(useLivePanelState.getState().availableItems).toEqual([
+      LivePanelItem.APPLICATION,
+      LivePanelItem.CHAT,
+    ]);
+  });
+
+  it('closes the panel and leave current item untuched', () => {
+    useLivePanelState.setState({
+      availableItems: [LivePanelItem.APPLICATION, LivePanelItem.CHAT],
+      currentItem: LivePanelItem.CHAT,
+      isPanelVisible: true,
+    });
+
+    //  update panel visibility with invalid item to select
+    useLivePanelState
+      .getState()
+      .setPanelVisibility(false, LivePanelItem.JOIN_DISCUSSION);
+
+    expect(useLivePanelState.getState().isPanelVisible).toBe(false);
+    expect(useLivePanelState.getState().currentItem).toEqual(
+      LivePanelItem.CHAT,
+    );
+    expect(useLivePanelState.getState().availableItems).toEqual([
+      LivePanelItem.APPLICATION,
+      LivePanelItem.CHAT,
+    ]);
+  });
+});
```

Comment:
I'd like to see another test such as setting a default currentItem not part of availableItems, what happens then?

Reply:
Isn't it done [here](https://github.com/openfun/marsha/pull/1229/files#diff-8849d66438fda8d4944044d0d726f7dce632928bec1b22bdb597b9618fc78439R96-R118) ?